src/main/java/com/school/project/service/impl/UserServiceImpl.java

```diff
@@ -1,17 +1,56 @@
 package com.school.project.service.impl;
 
+import com.school.project.dto.UserDto;
+import com.school.project.exception.UserNotFoundException;
+import com.school.project.model.AccountType;
+import com.school.project.model.User;
+import com.school.project.repository.UserAccountRepository;
+import com.school.project.repository.UserRepository;
+import com.school.project.service.UserAccountService;
 import com.school.project.service.UserService;
+import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Service;
 
+import java.util.Optional;
+
 @Service
 public class UserServiceImpl implements UserService {
+    @Autowired
+    UserRepository userRepository;
+    @Autowired
+    UserAccountService userAccountService;
+    @Autowired
+    UserAccountRepository userAccountRepository;
+
+
     @Override
-    public void create() {
+    public User create(User user) {
+        userRepository.save(user);
+        userAccountService.createUserAccount(user, AccountType.STUDENT);
+        return user;
+    }
 
+    @Override
+    public void update(UserDto user, Long id) {
+        User userById = getUserById(id);
+        userById.setFirstName(user.getFirstName());
+        userById.setLastName(user.getLastName());
+        userById.setBirthDate(user.getBirthDate());
+        userById.setEmail(user.getEmail());
+        userById.setPhoneNumber(user.getPhoneNumber());
+        userRepository.saveAndFlush(userById);
     }
 
     @Override
-    public void update() {
+    public void delete(Long id) {
+        userRepository.deleteById(id);
+        userAccountRepository.deleteById(id);
```

Comment:
are u sure u are deleting the right user account? just to be clear: u are deleting here user by it's id and then userAccount by it's id, not by user id. Are u 100% sure the user.id == userAccount.id??